widget/editor.go

```diff
@@ -223,8 +268,78 @@ func (e *Editor) processPointer(gtx layout.Context) {
 			if e.scroller.State() != gesture.StateFlinging {
 				e.caret.scroll = true
 			}
+
+			if evt.Modifiers == key.ModShift {
+				if e.IsSelected() {
```

Comment:
If you try it in other editors, you'll see that this is not so.  (At least the editor that I tried, to wit, the Github comment editor, ha; and also I just tried it in the Google Mail editor.)  Shift-Click near one side of the selection moves that side, and nearer the other moves that side.

Consider

```
aaa s|ss ddd
```

with the caret where the `|` is.  If you Shift-Click on the last d of ddd, that should select `ss ddd`, and move the caret to where you clicked.  If you then Shift-Click in front of the first a of aaa, that should select `aaa sss ddd` (and also move the caret to where you clicked).

If I do as you suggest, then the first Shift-Click does select `ss ddd`, but the second selects `aaa s`, which is wrong.  Or at least not what *I* want.

Reply:
I tried your example in the GH comment editor on macOS. It seems to me your description of the intended behaviour doesn't match:

https://user-images.githubusercontent.com/524812/103751349-70f50b80-5008-11eb-93c7-cd2013850775.mov

Regardless of the correct behaviour, I stand by my point to move the selection awareness to the lower-level Editor operations (move, delete etc.). In other words, I hope that the higher-level processPointer can remain unaware of whether there are a non-zero selection.

Reply:
Hahaha, it's good that you posted that video, because otherwise I think we'd've assumed we were talking past each other.  But nope, you and I actually see different behavior.

aaa sss ddd

https://user-images.githubusercontent.com/2324697/103779959-633c8780-5002-11eb-9752-604c8cbc7099.mov

> I stand by my point to move the selection awareness to the lower-level Editor operations

Sure, I'm fine with that.

I also stand by my point that my desired behavior is to move the selection endpoints depending on what you clicked closest to.  I like that behavior and I think it's most convenient for users.

That said, just to be clear, I'm willing to do it however you want me to do it.  :)  The current patch implements the "extend" behavior; give it a try and see what you think.  (Or, obviously, if you have already and know that you hate it, just tell me what you want, that's fine too.)

Reply:
The difference seems to be our browsers: Firefox has the move-endpoints behavior, while Chrome has your extend behavior. I suspect move-endpoints is consistent with Linux/Windows while extend is consistent with macOS. I tried macOS' TextEdit and it also extends.

Given the inconsistencies, and your preference, please do keep the extend behaviour. We can always add a runtime.GOOS check for the move-endpoints behaviour later.